nuc_morph_analysis/lib/preprocessing/add_features.py

```diff
@@ -140,6 +140,96 @@ def add_feature_at(df, frame_column, feature, feature_column, multiplier=1):
         )
     return df
 
+def add_mean_feature_over_trajectory(df, feature_list, multiplier_list):
+    """
+    Add the mean of a given feature over the growth trajectory 
+    from transition to frame breakdown. 
+    
+    Parameters
+    ----------
+    df : DataFrame
+        The dataframe
+    feature_list : list
+        List of column names
+    multiplier_list : list
+        List of scale to multiply the mean by
+        
+    Returns
+    -------
+    df : DataFrame
+        The dataframe with the added mean feature columns
+    """
+    for feature, multiplier in zip(feature_list, multiplier_list):
+        for tid, dft in df.groupby("track_id"):
+            start = dft.frame_transition.values[0]
+            stop = dft.Fb.values[0]
+            df_mean = dft[(dft['index_sequence'] >= start) & (dft['index_sequence'] <= stop)]
+            mean = df_mean[feature].mean() * multiplier
+            df.loc[df.track_id == tid, f"mean_{feature}"] = mean
+    return df
+
+
+def get_early_transient_gr_of_whole_colony(df, scale, time_shift=25):
```

Comment:
25 = 2 hr 5 min, but below you say this should be at 2 hr

Reply:
Yes! At first I did two hours but the feature was not available for all the tracks so you need to go to the next time point! Good catch.